# Tasca M2 T01


## Exercici 1

Crea una llista que agrupi els mesos de l’any en trimestres (1T: gener, febrer i març; 2T: abril, maig, juny...), és a dir, una llista amb 4 llistes dins.

In [1]:
listMeses = [["enero", "febrero", "marzo"],["abril", "mayo", "junio"],["julio", "agosto", "septiembre"],["octubre", "noviembre", "diciembre"]]

## Exercici 2

Crea un codi que et permeti accedir a:

1. El segon mes del primer trimestre.
2. Els mesos del primer trimestre.
3. Setembre i octubre.

### El segon mes del primer trimestre.


In [2]:
print(listMeses[0][1])

febrero


### Els mesos del primer trimestre.

In [3]:
 print(listMeses[0])

['enero', 'febrero', 'marzo']


### Setembre i octubre.

In [4]:
print(f'{listMeses[2][2]} y {listMeses[3][0]}') 

#Otras formas de printear
#print(listMeses[2][2],",", listMeses[3][0])   
#print("%s %s meses %s" % (listMeses[2][2], listMeses[3][0], 'hola')) 


septiembre y octubre


### Resultados 
1. El segundo mes del primer trimestre es: febrero
2. Los meses del primer trimestre son: ['enero', 'febrero', 'marzo']
3. septiembre y octubre


## Exercici 3

Crea una llista amb nombres desordenats i respon a les següents preguntes:

1. Quants números hi ha?
2. Quantes vegades apareix el número 3.
3. Quantes vegades apareixen els nombres 3 i 4?
4. Quin és el número més gran?
5. Quins són els 3 números més petits?
6. Quin és el rang d’aquesta llista?


### Crea una llista amb nombres desordenats i respon a les següents preguntes:

In [10]:
listNumeros = [3,56,7,34,4,23,78,4,234,12,3,67]
print(listNumeros)

[3, 56, 7, 34, 4, 23, 78, 4, 234, 12, 3, 67]


### Quants números hi ha?

In [11]:
print(len(listNumeros)) 

12


### Quantes vegades apareix el número 3.

In [12]:
print(listNumeros.count(3))

2


### Quantes vegades apareixen els nombres 3 i 4?

In [14]:
print(f'El número 3 aparece {listNumeros.count(3)} veces y el número 4 aparece {listNumeros.count(4)} veces.') 

El número 3 aparece 2 veces y el número 4 aparece 2 veces


### Quin és el número més gran?

In [15]:
print(max(listNumeros)) 

234


### Quins són els 3 números més petits?

In [28]:
import heapq
print(f'Los 3 números más pequeños son: {heapq.nsmallest(3,listNumeros)}')

Los 3 números más pequeños son: [3, 3, 4]


In [29]:
listNumeros.sort()
print(f'Lista ordenada: {listNumeros}')
print(f'Los 3 números más pequeños son: {listNumeros[0:3]}')

Lista ordenada: [3, 3, 4, 4, 7, 12, 23, 34, 56, 67, 78, 234]
Los 3 números más pequeños son: [3, 3, 4]


### Quin és el rang d’aquesta llista?

In [32]:
print(f'El rango de la lista es: {max(listNumeros) - min(listNumeros)}') 

El rango de la lista es: 231


### Resultados
1. Lista creada: [3, 56, 7, 34, 4, 23, 78, 4, 234, 12, 3, 67]
2. Tiene 12 números
3. El número 3 aparece 2 veces 
4. El número 3 y el número 4 aparecen 2 veces
5. El número más grande es el 234
6. Los números más pequeños son [3, 3, 4]
7. El rango de esta lista es: 231

## Excercici 4

Crea un diccionari de la següent forma i respon a les preguntes:

compra = { "Pomes" : {"Qty": 5, "€": 0.42}, "Peres" : {"Qty": 3, "€": 0.66} }

1. Afegeix alguna fruita més
2. Quant han costat les peres en total?
3. Quantes fruites hem comprat en total?
4. Quina és la fruita més cara?

### Afegeix alguna fruita més

In [31]:
compra = { 
    "Pomes" : {"Qty": 5, '€': 0.42}, 
    "Peres" : {"Qty": 3, '€': 0.66} }

compra ["Taronges"] = {"Qty": 4, '€': 0.56} 
print(compra["Taronges"])
print(compra)

{'Qty': 4, '€': 0.56}
{'Pomes': {'Qty': 5, '€': 0.42}, 'Peres': {'Qty': 3, '€': 0.66}, 'Taronges': {'Qty': 4, '€': 0.56}}


### Quant han costat les peres en total?

In [25]:
print(f'Las peras en total costaron: €{compra["Peres"]["Qty"] * compra ["Peres"]["€"]}')


Las peras en total costaron: €1.98


### Quantes fruites hem comprat en total?

In [17]:
totalFrut = sum([x["Qty"] for x in compra.values()])
print(f'Han comprado en total {totalFrut} frutas.')

 Han comprado en total 12 frutas


### Quina és la fruita més cara?

In [23]:
frutCara = max(x["€"]for x in compra.values())
for key,value in compra.items():
    if value["€"] == frutCara:          
        print(f'La fruta más cara es: {key}')

 La fruta más cara es: Peres


### Resultados 
1. Lista  con una fruta más agregada: compra = {'Pomes': {'Qty': 5, '€': 0.42}, 'Peres': {'Qty': 3, '€': 0.66}, 'Toronjas': {'Qty': 4, '€': 0.56}}
2. Las peras en total costaron:  1.98
3. Han comprado en total 12 frutas
4. La fruta más cara es: Peres 
